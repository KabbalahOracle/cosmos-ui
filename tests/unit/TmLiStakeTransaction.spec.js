import { mount } from "@vue/test-utils"
import { TmLiStakeTransaction } from "@/index"
import transactions from "../../assets/txs.js"

describe("TmLiStakeTransaction", () => {
  let wrapper
  let propsData = {
    transaction: transactions[3],
    validators: [
      {
        owner: "cosmosvaladdr15ky9du8a2wlstz6fpx3p4mqpjyrm5ctqzh8yqw",
        description: {
          moniker: "cool validator"
        },
        tokens: "20",
        delegator_shares: "100"
      },
      {
        owner: "cosmosvaladdr157mg9hnhchfrqvk3enrvmvj29yhmlwf759xrgw",
        description: {
          moniker: "Kentucky val"
        },
        tokens: "10",
        delegator_shares: "40"
      }
    ],
    validatorURL: "/validator",
    bondingDenom: "stake",
    sharesToTokens: undefined
  }

  beforeEach(() => {
    wrapper = mount(TmLiStakeTransaction, { propsData })
  })

  describe(`delegations`, () => {
    it("should show delegations", () => {
      expect(wrapper.vm.delegation).toBe(true)
      expect(wrapper.vm.$el).toMatchSnapshot()
    })
  })

  describe(`unbonding delegations`, () => {
    it("should show unbondings", () => {
      wrapper.setProps({
        transaction: transactions[4]
      })
      expect(wrapper.vm.unbonding).toBe(true)
      expect(wrapper.vm.$el).toMatchSnapshot()
    })

    it("should show unbondings by displaying the given token amount", () => {
      wrapper.setProps({
        transaction: transactions[4],
        sharesToTokens: 15
      })
      expect(wrapper.vm.unbonding).toBe(true)
      expect(wrapper.vm.$el).toMatchSnapshot()
    })
  })

  describe(`redelegations`, () => {
    it("should show redelegations and calculate tokens from shares", () => {
      wrapper.setProps({
        transaction: transactions[5]
      })
      expect(wrapper.vm.redelegation).toBe(true)
      expect(wrapper.vm.$el).toMatchSnapshot()
    })

    it("should show redelegations by displaying the given token amount", () => {
      wrapper.setProps({
        transaction: transactions[5],
        sharesToTokens: 15
      })
      expect(wrapper.vm.redelegation).toBe(true)
      expect(wrapper.vm.$el).toMatchSnapshot()
    })
  })
})
