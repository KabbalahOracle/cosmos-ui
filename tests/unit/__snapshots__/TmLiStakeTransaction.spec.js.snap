// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TmLiStakeTransaction delegations should show delegations 1`] = `
<div
  class="tm-li-tx"
>
  <div
    class="tm-li-tx__icon"
  >
    <img
      src="../../assets/cosmos-logo.png"
      style="border-color: #512584;"
    />
  </div>
  <div
    class="tm-li-tx__content"
  >
    <div
      class="tm-li-tx__content__left"
    >
      <div
        class="tm-li-tx__content__caption"
      >
        <p
          class="tm-li-tx__content__caption__title"
        >
          <div>
            Delegated 
            <b>
              4200
            </b>
            <span>
               stakes
            </span>
          </div>
        </p>
      </div>
      <div
        class="tm-li-tx__content__information"
      >
        <div>
          To 
          <router-link
            to="/validator/cosmosvaladdr15ky9du8a2wlstz6fpx3p4mqpjyrm5ctqzh8yqw"
          >
            cool validator
          </router-link>
        </div>
      </div>
    </div>
    <!---->
    <div
      class="tm-li-tx__content__block"
    >
      <a
        href="https://explorecosmos.network/blocks/566"
      >
        Block #566 
      </a>
      @ 2:03 am
    </div>
  </div>
</div>
`;

exports[`TmLiStakeTransaction redelegations should show redelegations and calculate tokens from shares 1`] = `
<div
  class="tm-li-tx"
>
  <div
    class="tm-li-tx__icon"
  >
    <img
      src="../../assets/cosmos-logo.png"
      style="border-color: #7227b1;"
    />
  </div>
  <div
    class="tm-li-tx__content"
  >
    <div
      class="tm-li-tx__content__left"
    >
      <div
        class="tm-li-tx__content__caption"
      >
        <p
          class="tm-li-tx__content__caption__title"
        >
          <div>
            Redelegated 
            <b>
              0
            </b>
            <span>
               stakes
            </span>
          </div>
        </p>
      </div>
      <div
        class="tm-li-tx__content__information"
      >
        <div>
          From 
          <router-link
            to="/validator/cosmosvaladdr15ky9du8a2wlstz6fpx3p4mqpjyrm5ctqzh8yqw"
          >
            cool validator
          </router-link>
           to 
          <router-link
            to="/validator/cosmosvaladdr157mg9hnhchfrqvk3enrvmvj29yhmlwf759xrgw"
          >
            Kentucky val
          </router-link>
        </div>
      </div>
    </div>
    <!---->
    <div
      class="tm-li-tx__content__block"
    >
      <a
        href="https://explorecosmos.network/blocks/566"
      >
        Block #566 
      </a>
      @ 2:03 am
    </div>
  </div>
</div>
`;

exports[`TmLiStakeTransaction redelegations should show redelegations by displaying the given token amount 1`] = `
<div
  class="tm-li-tx"
>
  <div
    class="tm-li-tx__icon"
  >
    <img
      src="../../assets/cosmos-logo.png"
      style="border-color: #7227b1;"
    />
  </div>
  <div
    class="tm-li-tx__content"
  >
    <div
      class="tm-li-tx__content__left"
    >
      <div
        class="tm-li-tx__content__caption"
      >
        <p
          class="tm-li-tx__content__caption__title"
        >
          <div>
            Redelegated 
            <b>
              15
            </b>
            <span>
               stakes
            </span>
          </div>
        </p>
      </div>
      <div
        class="tm-li-tx__content__information"
      >
        <div>
          From 
          <router-link
            to="/validator/cosmosvaladdr15ky9du8a2wlstz6fpx3p4mqpjyrm5ctqzh8yqw"
          >
            cool validator
          </router-link>
           to 
          <router-link
            to="/validator/cosmosvaladdr157mg9hnhchfrqvk3enrvmvj29yhmlwf759xrgw"
          >
            Kentucky val
          </router-link>
        </div>
      </div>
    </div>
    <!---->
    <div
      class="tm-li-tx__content__block"
    >
      <a
        href="https://explorecosmos.network/blocks/566"
      >
        Block #566 
      </a>
      @ 2:03 am
    </div>
  </div>
</div>
`;

exports[`TmLiStakeTransaction unbonding delegations should show unbondings 1`] = `
<div
  class="tm-li-tx"
>
  <div
    class="tm-li-tx__icon"
  >
    <img
      src="../../assets/cosmos-logo.png"
      style="border-color: #9d5c19;"
    />
  </div>
  <div
    class="tm-li-tx__content"
  >
    <div
      class="tm-li-tx__content__left"
    >
      <div
        class="tm-li-tx__content__caption"
      >
        <p
          class="tm-li-tx__content__caption__title"
        >
          <div>
            Unbonded 
            <b>
              20
            </b>
            <span>
               stakes
            </span>
            <!---->
          </div>
        </p>
      </div>
      <div
        class="tm-li-tx__content__information"
      >
        <div>
          From 
          <router-link
            to="/validator/cosmosvaladdr15ky9du8a2wlstz6fpx3p4mqpjyrm5ctqzh8yqw"
          >
            cool validator
          </router-link>
        </div>
      </div>
    </div>
    <div
      class="tm-li-tx__content__action"
    >
      <div>
        <!---->
      </div>
    </div>
    <div
      class="tm-li-tx__content__block"
    >
      <a
        href="https://explorecosmos.network/blocks/566"
      >
        Block #566 
      </a>
      @ 2:03 am
    </div>
  </div>
</div>
`;

exports[`TmLiStakeTransaction unbonding delegations should show unbondings by displaying the given token amount 1`] = `
<div
  class="tm-li-tx"
>
  <div
    class="tm-li-tx__icon"
  >
    <img
      src="../../assets/cosmos-logo.png"
      style="border-color: #9d5c19;"
    />
  </div>
  <div
    class="tm-li-tx__content"
  >
    <div
      class="tm-li-tx__content__left"
    >
      <div
        class="tm-li-tx__content__caption"
      >
        <p
          class="tm-li-tx__content__caption__title"
        >
          <div>
            Unbonded 
            <b>
              15
            </b>
            <span>
               stakes
            </span>
            <!---->
          </div>
        </p>
      </div>
      <div
        class="tm-li-tx__content__information"
      >
        <div>
          From 
          <router-link
            to="/validator/cosmosvaladdr15ky9du8a2wlstz6fpx3p4mqpjyrm5ctqzh8yqw"
          >
            cool validator
          </router-link>
        </div>
      </div>
    </div>
    <div
      class="tm-li-tx__content__action"
    >
      <div>
        <!---->
      </div>
    </div>
    <div
      class="tm-li-tx__content__block"
    >
      <a
        href="https://explorecosmos.network/blocks/566"
      >
        Block #566 
      </a>
      @ 2:03 am
    </div>
  </div>
</div>
`;
